sky/provision/lambda_cloud/instance.py

```diff
@@ -248,18 +249,70 @@ def query_instances(
     return statuses
 
 
-def open_ports(
-    cluster_name_on_cloud: str,
-    ports: List[str],
-    provider_config: Optional[Dict[str, Any]] = None,
-) -> None:
-    raise NotImplementedError('open_ports is not supported for Lambda Cloud')
+def open_ports(cluster_name_on_cloud: str,
+               ports: List[str],
+               provider_config: Optional[Dict[str, Any]] = None) -> None:
+    """Open ports for a cluster.
+
+    Args:
+        cluster_name_on_cloud: The cluster name (unused as rules are global).
+        ports: The ports to open.
+        provider_config: The provider config (unused).
+    """
+    del cluster_name_on_cloud, provider_config  # Unused.
+
+    # Lambda Cloud firewall rules are global (not cluster-specific)
+    # Initialize Lambda Cloud client
+    lambda_client = lambda_utils.LambdaCloudClient()
+
+    # Get existing firewall rules
+    existing_rules = lambda_client.list_firewall_rules()
+
+    # Extract existing ports
+    existing_ports: Set[int] = set()
+    for rule in existing_rules:
+        if rule.get('protocol') == 'tcp':
+            port_range = rule.get('port_range')
+            if port_range and len(port_range) == 2:
+                # If it's a single port (same min and max)
+                if port_range[0] == port_range[1]:
+                    existing_ports.add(port_range[0])
+                else:
+                    # For port ranges, add all ports in the range
+                    existing_ports.update(
+                        range(port_range[0], port_range[1] + 1))
+
+    # Convert port strings to a set of individual ports
+    ports_to_open = resources_utils.port_ranges_to_set(ports)
+
+    # Remove ports that are already open
+    ports_to_open = ports_to_open - existing_ports
+
+    # Open new ports
+    for port in ports_to_open:
+        logger.info(f'Opening port {port}/tcp')
```

Comment:
Would it be useful to have smarter logic here to consolidate contiguous ports numbers into a single firewall rule? E.g., if. `ports_to_open` is 8000-8100, we should probably create one rule with the range instead of creating 100 rules.

Ok to leave as a TODO.

Reply:
Good point! Fixed. Thanks!